### PR TITLE
Bug 2039359: Fix adm prune rs orphans

### PR DESCRIPTION
**Pull request by deejross**
This allows `oc adm prune deployments --replica-sets=true --orphans=true` to work on orphaned ReplicaSets.

### CODE REVIEW

File: pkg/cli/admin/prune/deployments/prune.go [hunk 1]
```diff
@@ -60,11 +60,11 @@ func NewPruner(options PrunerOptions) Pruner {
 
 	filter := &andFilter{
 		filterPredicates: []FilterPredicate{
-			FilterDeploymentsPredicate,
```
Comment (soltysh) [inline pkg/cli/admin/prune/deployments/prune.go:63]: Are you sure you want to entirely remove that filter? Shouldn't that depend on the `Orphans` option?

Comment (deejross) [reply pkg/cli/admin/prune/deployments/prune.go:63]: Yes, that's what made it work. That filter ignores the orphans option completely.

Comment (soltysh) [reply pkg/cli/admin/prune/deployments/prune.go:63]: What I'm saying is, that depending if `--orphans` flag was specified we should add the filter or not.

File: pkg/cli/admin/prune/deployments/prune_test.go [hunk 2]
```diff
@@ -61,14 +61,17 @@ func TestPruneTask(t *testing.T) {
...
+			deployment := mockDeployment("b", "deployment")
+			deployments = append(deployments, deploymentConfig, deployment)
 
 			replicas = append(replicas, withCreated(withStatus(mockReplicationController("a", "build-1", deploymentConfig), deploymentStatusOption), now))
 			replicas = append(replicas, withCreated(withStatus(mockReplicationController("a", "build-2", deploymentConfig), deploymentStatusOption), old))
 			replicas = append(replicas, withSize(withCreated(withStatus(mockReplicationController("a", "build-3-with-replicas", deploymentConfig), deploymentStatusOption), old), 4))
 			replicas = append(replicas, withCreated(withStatus(mockReplicationController("a", "orphan-build-1", nil), deploymentStatusOption), now))
 			replicas = append(replicas, withCreated(withStatus(mockReplicationController("a", "orphan-build-2", nil), deploymentStatusOption), old))
 			replicas = append(replicas, withSize(withCreated(withStatus(mockReplicationController("a", "orphan-build-3-with-replicas", nil), deploymentStatusOption), old), 4))
+			replicas = append(replicas, mockReplicaSet("b", "rs1", deployment))
```
Comment (soltysh) [inline pkg/cli/admin/prune/deployments/prune_test.go:73]: This test works fine with `FilterDeploymentsPredicate`, but you're removing that filter above, why?

File: pkg/cli/admin/prune/deployments/prune_test.go [hunk 3]
```diff
@@ -34,83 +34,167 @@ func (m *mockDeleteRecorder) Verify(t *testing.T, expected sets.String) {
...
+			KeepComplete:    10,
+			KeepFailed:      0,
+			ExpectedPruneNames: []string{
+				"build-3",
+				"orphan-build-1",
+				"orphan-build-2",
+				"orphan-rs1",
+			},
+		},
+		"prune keep 2 non-orphaned, completed relpicas": {
```
Comment (soltysh) [inline pkg/cli/admin/prune/deployments/prune_test.go:140]: typo: replicas

File: pkg/cli/admin/prune/deployments/prune_test.go [hunk 4]
```diff
@@ -34,83 +34,167 @@ func (m *mockDeleteRecorder) Verify(t *testing.T, expected sets.String) {
...
+			ReplicaSets:     true,
+			KeepToungerThan: 0,
+			KeepComplete:    2,
+			KeepFailed:      0,
+			ExpectedPruneNames: []string{
+				"build-3",
+				"rs1",
+			},
+		},
+		"prune keep 1 non-orphaned, failed relpicas and 2 non-orphaned, completed replicas": {
```
Comment (soltysh) [inline pkg/cli/admin/prune/deployments/prune_test.go:151]: same typo: replicas

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**